logging.go

```diff
@@ -61,14 +61,17 @@ func (h loggingHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 }
 
 func makeLogger(w http.ResponseWriter) (*responseLogger, http.ResponseWriter) {
-	logger := &responseLogger{w: w, status: http.StatusOK}
+	logger := &responseLogger{w: w, status: 0}
 	return logger, httpsnoop.Wrap(w, httpsnoop.Hooks{
 		Write: func(httpsnoop.WriteFunc) httpsnoop.WriteFunc {
 			return logger.Write
 		},
 		WriteHeader: func(httpsnoop.WriteHeaderFunc) httpsnoop.WriteHeaderFunc {
 			return logger.WriteHeader
 		},
+		Hijack: func(hijackFunc httpsnoop.HijackFunc) httpsnoop.HijackFunc {
```

Comment:
@astrolox I haven't dug into it but an initial gloss seems no change is needed on line 64 and instead a new line here in this function should be:
```
logger.status = 0
```
does the trick. There may be some subtleties I'm missing, however.

Reply:
Logically. I agree that would do the trick.
Also it's a less risky change, so I understand why that might be preferable.

However, I would not set it to 0 within the hijack function, because on reading back the code a plain english description of it's behaviour is
1. setting a default
2. unsetting the default
3. checking to see if a default has been set https://github.com/gorilla/handlers/blob/3e030244b4ba0480763356fc8ca0ade6222e2da0/handlers.go#L75

My view is that this isn't ideal from a clean code perspective.

My preference is to simply not set a default in the first place. This then restores the assumption (which I was presume was made by the author of the code at line 75 of handlers.go) that the value would be 0 until it is set from outside the logger.

If we would prefer not to remove the default (i.e. less risk), then I suggest we should use the hijack method to change the status to 101 and remove lines 75 to 78 in handlers.go which set the status to 101 only if it's not set. Again my reasoning is because by having a default it can never be not set, so checking for that seems wrong.

Reply:
It is an interesting quirk there's this setting of a default status value. If I were hesitate to guess, it'd be that the default behavior in the std lib is to write `200` if no header is written by a handler and that may not necessitate a call to the `WriteHeader` method for `ResponseWriter` the `responseLogger` wraps. That sounds a bit strange if it were the case.

As well, it's a bit odd that `*responseLogger.Hijack` is making this assumption about a status code in the first place and so removing the setting of a default makes the most sense to me, assuming a properly implemented server always calls `WriteHeader`.

I don't see unit tests covering the code under discussion so some preliminary ones identifying the current behavior may illuminate what to do here. Do you have the time for that or would you like some assistance?